exporters/otlp/otlptrace/otlptracehttp/client.go

```diff
@@ -165,7 +165,7 @@ func (d *client) UploadTraces(ctx context.Context, protoSpans []*tracepb.Resourc
 		}
 
 		switch resp.StatusCode {
-		case http.StatusOK:
+		case http.StatusOK, http.StatusAccepted:
```

Comment:
We should accept all 2xx codes, as per HTTP specification. see https://github.com/open-telemetry/opentelemetry-specification/blob/v1.16.0/specification/protocol/otlp.md#json-protobuf-encoding . See what we do in OTLP exporter in collector https://github.com/open-telemetry/opentelemetry-collector/blob/892d07dffbb8dc639114bc0c340dfd4fc5293258/exporter/otlphttpexporter/otlp.go#L146-L149

Reply:
> On success, the server MUST respond with `HTTP 200 OK`.

> If the request is only partially accepted (i.e. when the server accepts only parts of the data and rejects the rest), the server MUST respond with `HTTP 200 OK`.

Nothing other than a `200 OK` response is according to spec.

Reply:
Good call, updated